sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -85,6 +86,31 @@ def _update_leaves_values(loss, grower, y_true, raw_prediction, sample_weight):
 class BaseHistGradientBoosting(BaseEstimator, ABC):
     """Base class for histogram-based gradient boosting estimators."""
 
+    _parameter_constraints = {
+        "loss": [BaseLoss],
+        "learning_rate": [Interval(Real, 0, None, closed="neither")],
+        "max_iter": [Interval(Integral, 1, None, closed="left")],
+        "max_leaf_nodes": [Interval(Integral, 2, None, closed="left"), None],
+        "max_depth": [Interval(Integral, 1, None, closed="left"), None],
+        "min_samples_leaf": [Interval(Integral, 1, None, closed="left")],
+        "l2_regularization": [Interval(Real, 0, None, closed="left")],
```

Comment:
These params were originally validated in `TreeGrower`. If these are not specified or specified in `TreeGrower`, it will not pass the common check test.

Reply:
Since TreeGrower is only used by HistGB estimators, and is not a public class, I think it would make sense to remove these validations from TreeGrower and rely on the validation of HGB estimators

Reply:
TreeGrow is an internal class only useful for code factorization and is not meant to be a general purpose estimator (no `score` method for instance).